src/java/yaw/engine/items/HitBox.java

```diff
@@ -0,0 +1,65 @@
+package yaw.engine.items;
+
+import org.joml.Vector3f;
+import org.joml.Vector4f;
+import yaw.engine.collision.Collision;
+import yaw.engine.meshs.Mesh;
+
+import java.util.ArrayList;
+
+/**
+ * HitBox is an ItemObject, but unlike a classical ItemObject we can check if there is a collision with other Hitboxes
+ */
+public class HitBox extends ItemObject {
+    /**
+     *
+     * @param pId           Name of the HitBox
+     * @param pRotation     Initial Rotation
+     * @param pPosition     Initial Position
+     * @param pScale        HitBox scale
+     * @param mesh          Mesh associated to the HitBox
+     */
+    public HitBox(String pId, Vector3f pRotation, Vector3f pPosition, float pScale, Mesh mesh){
+        super(pId,pRotation,pPosition,pScale,mesh);
+    }
+
+
+    /**    Testing the collision with another HitBox
+     *
+     * @param item      the second hitbox to test collision with
+     * @return          true if item is touching the hitbox which is calling the method
+     */
+    public boolean isIsCollisionWith(HitBox item){
```

Comment:
Do you think it's worthy to inline the computation ?  (don't know if javac is doing the optim')